samples/SkiaJsSample/src/jsMain/kotlin/org/jetbrains/skiko/sample/js/App.kt

```diff
@@ -2,46 +2,127 @@ package org.jetbrains.skiko.sample.js
 
 import kotlinx.browser.document
 import kotlinx.browser.window
-import org.jetbrains.skia.*
-import org.jetbrains.skiko.wasm.GetWebGLContext
+import org.jetbrains.skia.Canvas
+import org.jetbrains.skia.Color4f
+import org.jetbrains.skia.Paint
+import org.jetbrains.skia.PaintMode
+import org.jetbrains.skia.Rect
+import org.jetbrains.skiko.wasm.api.CanvasRenderer
 import org.jetbrains.skiko.wasm.onWasmReady
 import org.w3c.dom.HTMLCanvasElement
+import kotlin.math.PI
+import kotlin.math.cos
+import kotlin.math.sin
 
+private data class Circle(var x: Float, var y: Float, var r: Float)
 
-fun main() {
-    onWasmReady {
-        val paint = Paint().apply {
-            color4f = Color4f(0f, 1f, 0f, 1.0f)
-            mode = PaintMode.FILL
-            isAntiAlias = true
+private fun Canvas.draw(circle: Circle, paint: Paint) {
+    drawCircle(circle.x, circle.y, circle.r, paint)
+}
+
+private fun Circle.move(s: Float, angle: Float) {
+    x += s * sin(angle)
+    y += s * cos(angle)
+}
+
+private enum class Position {
+    INSIDE,
+    TOUCHES_SOUTH,
+    TOUCHES_NORTH,
+    TOUCHES_WEST,
+    TOUCHES_EAST
+}
+
+private fun calculatePosition(circle: Circle, boundingWidth: Int, boundingHeight: Int): Position {
+    val southmost = circle.y + circle.r
+    val northmost = circle.y - circle.r
+    val westmost = circle.x - circle.r
+    val eastmost = circle.x + circle.r
+
+    return when {
+        southmost >= boundingHeight -> Position.TOUCHES_SOUTH
+        northmost <= 0 -> Position.TOUCHES_NORTH
+        eastmost >= boundingWidth -> Position.TOUCHES_EAST
+        westmost <= 0 -> Position.TOUCHES_WEST
+        else -> Position.INSIDE
+    }
+}
+
+private class BouncingBall(
+    val circle: Circle,
+    val width: Int,
+    val height: Int,
+    val velocity: Float,
+    var angle: Double
+) {
+    fun recalculate(dt: Float) {
+        val position = calculatePosition(circle, width, height)
+
+        when (position) {
```

Comment:
Make when exhaustive.

Reply:
well, I can add else or check for inside but in this case angle just doesn't change at all, we basically do nothing in that case.

Reply:
Kotlin compiler warns on non-exhaustive checks, enough to add INSIDE.